src/naive/isoweek.rs

```diff
@@ -47,7 +47,7 @@ pub(super) fn iso_week_from_yof(year: i32, of: Of) -> IsoWeek {
         }
     };
     let flags = YearFlags::from_year(year);
-    IsoWeek { ywf: (year << 10) | (week << 4) as DateImpl | DateImpl::from(flags.0) }
+    IsoWeek { ywf: DateImpl::new((year << 10) | (week << 4) as i32 | flags.0 as i32) }
```

Comment:
Maybe we need `DateImpl::from_yw()` or `DateImpl::from_yof()`? Looks like we have a bunch of repetition of similar patterns.